File: php/src/WaitException.php
<?php

declare(strict_types=1);

namespace Idkollen\Client;

class WaitException extends \RuntimeException
{
    public function __construct(
        public readonly bool $timeout,
        ?\Throwable $previous = null,
    ) {
        parent::__construct(
            $timeout ? 'Poll timed out' : 'Poll error: ' . ($previous?->getMessage() ?? ''),
            0,
            $previous,
        );
    }
}

File: csharp/src/Idkollen.Client/WaitException.cs
namespace Idkollen.Client;

public class WaitException : Exception
{
    public bool Timeout { get; }

    public WaitException(bool timeout, Exception? innerException = null)
        : base(timeout ? "Poll timed out" : "Poll error", innerException)
    {
        Timeout = timeout;
    }
}
